### ISSUE TITLE
Make Variable Explorer column headers movable

### ISSUE THREAD

**Issue by battaglia01** [spyder-ide/spyder#16423]
The variable explorer currently has the columns as Name -> Type -> Size -> Value, which is kind of annoying as the main thing I usually care about is Value which is way off at the end. There is no way to change this order but a simple one-liner in Qt could make the column headers draggable. In `collectionseditor.py`, in class `BaseTableView`'s `def setup_table(self)`, simply add the one line

```
    def setup_table(self):
        """Setup table"""
        self.horizontalHeader().setStretchLastSection(True)
        self.horizontalHeader().setSectionsMovable(True) # <--- NEW LINE HERE
        self.adjust_columns()
        # Sorting columns
        self.setSortingEnabled(True)
        self.sortByColumn(0, Qt.AscendingOrder)
```

and you're done. Note this is for BaseTableView; you could also do this instead in RemoteCollectionsEditorTableView and CollectionsEditorTableView (not sure which subclass is the best to do this in, or what other things subclass BaseTableView).

FWIW, I'd much prefer it if the default were changed to Name -> Value -> Type -> Size, but that seems more involved and I'd be happy enough with this.

## Versions

Spyder: 5.0.5 | Python 3.9.5 64-bit | Qt 5.9.7 | PyQt5 5.9.2 | Darwin 17.7.0

**Comment by steff456** [comment]
Hi @battaglia01,

This is something we need to discuss with the team, so @spyder-ide/core-developers what do you think about this? I'm afraid that if the value of the variable is too long, then the other columns will not be that visible. And I'm not sure if we want to enable those columns to be movable.

**Comment by dalthviz** [comment]
I think there is a restriction in the size that the columns can have at the beginning except for the last one which takes all the remaining available space (unless the user resizes manually the columns). However I think too that changing the default order of the columns could be quite involved (and the current order is probably like that to have more space assigned to the value column by default). Regarding being able to move the columns, I don't have a strong opinion, but maybe it could be handy for some people and also I think the object explorer allows you to move the columns already.

**Comment by ccordoba12** [comment]
What if we allow to hide certain columns instead of making them draggable?

**Comment by battaglia01** [comment]
What exactly is wrong with making the columns movable? I patched it in on
my Spyder and haven't seen any problem.

**Comment by ccordoba12** [comment]
I think there are several hard-coded things that assume the current column ordering.

**Comment by battaglia01** [comment]
I suggested `setSectionsMovable` because it seems to still internally handle column IDs correctly. e.g. if you drag the "Value" column to position 2, Qt still knows that it was originally the fourth column for when you add new variables into the explorer. But if it's easier to hide columns instead then alright, I'd be happy with that...

**Comment by ccordoba12** [comment]
Ok, if that's the case, then please submit a pull request about it.

**Comment by battaglia01** [comment]
Alright, although I will wait a little to hear if there are any additional comments from @steff456 and @dalthviz.

BTW: is there some way to get some of these top-level Spyder widgets, from within Spyder, at a running kernel? Something like `get_ipython()` but for the QApplication? I've tried `QtWidgets.QApplication.instance()` but sometimes that returns `NoneType`; also `spyder_kernels.customize.spydercustomize.SpyderQApplication._instance_list` often is empty...

**Comment by mrclary** [comment]
> BTW: is there some way to get some of these top-level Spyder widgets, from within Spyder, at a running kernel?

I'm not sure that I understand your question. Do you mean for debugging purposes?

**Comment by battaglia01** [comment]
Right.

**Comment by mrclary** [comment]
The only way that I know of is to access them from inside a namespace in which they reside.
Do you have a specific example of what you want to look at?
My general procedure is this:
1. Insert `import pdb; pdb.set_trace()` somewhere in the source code that I want to probe
2. launch spyder from commandline using bootstrap and some debug level
  ```bash
  (dev) $ python bootstrap.py -- --debug-info=miminal
  ```
3. My terminal will now show a `(pdb)` prompt that I can use for debugging.

Caveat: depending on where you place your `set_trace` there may be a Qt event loop running, in which case it will complain at you in the stdout of your terminal, rendering debugging impossible. To avoid this, you can place the following code before your `set_trace`.
```python
from qtpy.QtCore import pyqtRemoveInputHook; pyqtRemoveInputHook()
``` 

Does anyone have a better way of doing it? If so, I'd like to here how others do it.

**Comment by battaglia01** [comment]
No specific example, just sometimes I like to try various things, like in this case `setSectionsMovable`, without having to restart Spyder. It would be nice if there were some way to introspect Spyder itself from within the running kernel - i.e. something like getting the current application from `QtWidgets.QApplication.instance()`, and from there I should theoretically be able to look at `QtWidgets.QApplication.instance().topLevelWidgets()` or whatever, but since the kernel is in a different process, it doesn't work.

**Comment by mrclary** [comment]
I see. In that case, I would recommend running the widget by itself so you don't have to run the entire Spyder application. One example of how to do this is in the PYTHONPATH Manager widget: spyder/widgets/pathmanager.py. There you will find a `test` function and `if __name__ == '__main__'` protection block at the end of the module, so that you can run just this file as a script to play around with the widget without having to launch Spyder. There may be other examples throughout the source code, but this is one that I remember seeing recently. If the module that contains the widget you're interested in doesn't already have something like that, then try writing it and pushing it up so it can be used by others in the future!

Hope that helps.

**Comment by dalthviz** [comment]
> Alright, although I will wait a little to hear if there are any additional comments from @steff456 and @dalthviz.

I think is ok :+1: I also tried a little bit setting the movable columns in my local and seems like everything is working (as @battaglia01 says even if the columns move seems like their id remains the same)

**Comment by ccordoba12** [comment]
@battaglia01, if you want to tinker with Spyder internals at runtime, you can do it with the Internal Console, which can be opened by going to the menu `View > Panes`.

It exposes an object called `spy`, which holds a reference to the main window:

![imagen](https://user-images.githubusercontent.com/365293/134035230-91eee6e9-1045-487f-a66b-01fefa751048.png)

(just ignore the warning shown there).

Through it you can access any plugin and then alter its attributes:

![imagen](https://user-images.githubusercontent.com/365293/134036084-900eca2d-7db8-4762-b2df-34369b080f84.png)

![imagen](https://user-images.githubusercontent.com/365293/134035793-33d6b203-6910-4543-a4e5-9df82d88cbbd.png)

**Comment by steff456** [comment]
> Alright, although I will wait a little to hear if there are any additional comments from @steff456 and @dalthviz.

It sounds ok! Please submit the PR :)

**Comment by battaglia01** [comment]
Thank you all! I will submit when I get the chance.

**Comment by battaglia01** [comment]
OK, implemented in #17022.
